tests/system-tests/rdscore/internal/rdscorecommon/sriov-rootless-dpdk.go

```diff
@@ -0,0 +1,861 @@
+	dpdkServerMac            = "60:00:00:00:00:04"
+	serverPodLabel           = "rds-app=rootless-dpdk-server"
+	maxMulticastNoiseRate    = 5000
+	minimumExpectedDPDKRate  = 1000000
+
+	tapOneInterfaceName            = "ext0"
+	tapTwoInterfaceName            = "ext1"
+	tapThreeInterfaceName          = "ext2"
+	firstInterfaceBasedOnTapTwo    = "ext1.1"
+	secondInterfaceBasedOnTapTwo   = "ext1.2"
+	firstInterfaceBasedOnTapThree  = "ext2.1"
+	secondInterfaceBasedOnTapThree = "ext2.2"
+)
+
+var (
+	deploymentNamespace          = RDSCoreConfig.RootlessDPDKNamespace
+	dpdkVlanID                   = RDSCoreConfig.RootlessDPDKVlanID
+	dummyVlanID                  = RDSCoreConfig.RootlessDPDKDummyVlanID
+	dpdkDeploymentSAName         = RDSCoreConfig.RootlessDPDKDeploymentSA
+	firstInterfaceBasedOnTapOne  = fmt.Sprintf("%s.%s", tapOneInterfaceName, dpdkVlanID)
+	secondInterfaceBasedOnTapOne = fmt.Sprintf("%s.%s", tapOneInterfaceName, dummyVlanID)
+	dpdkNetworkOne               = RDSCoreConfig.RootlessDPDKNetworkOne
+	dpdkNetworkTwo               = RDSCoreConfig.RootlessDPDKNetworkTwo
+	dpdkPolicyTwo                = RDSCoreConfig.RootlessDPDKPolicyTwo
+	dpdkClientDeploymentName     = RDSCoreConfig.RootlessDPDKClientDeploymentName
+	dpdkClientVlanMac            = RDSCoreConfig.RootlessDPDKClientVlanMac
+	dpdkClientMacVlanMac         = RDSCoreConfig.RootlessDPDKClientMacVlanMac
+	dpdkClientIPVlanMac          = RDSCoreConfig.RootlessDPDKClientIPVlanMac
+	dpdkClientIPVlanIP           = RDSCoreConfig.RootlessDPDKClientIPVlanIPv4
+	dpdkClientIPVlanIPDummy      = RDSCoreConfig.RootlessDPDKClientIPVlanIPv4Dummy
+
+	rootUser         = int64(0)
+	waitTimeout      = 3 * time.Minute
+	psiWaitTimeout   = 10 * time.Second
+	psiRetryInterval = time.Second
+
+	serverDeploymentLabelMap = map[string]string{"rds-app": "rootless-dpdk-server"}
+
+	serverSC = corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_ADMIN", "NET_RAW"},
+		},
+	}
+)
+
+func createRootlessDPDKServerDeployment(
+	apiClient *clients.Settings,
+	serverSRIOVNetworkName,
+	clientMac,
+	sriovNodePolicyName,
+	serverDeploymentHost,
+	txIPs string) error {
+	glog.V(100).Infof("Ensuring server deployment %q doesn't exist in %q namespace",
+		serverDPDKDeploymentName, deploymentNamespace)
+
+	err := cleanUpRootlessDPDKDeployment(apiClient, serverDPDKDeploymentName, deploymentNamespace, serverPodLabel)
+	if err != nil {
+		glog.V(100).Infof("Failed to cleanup deployment %s from the namespace %s: %v",
+			serverDPDKDeploymentName, deploymentNamespace, err)
+
+		return fmt.Errorf("failed to cleanup deployment %s from the namespace %s: %w",
+			serverDPDKDeploymentName, deploymentNamespace, err)
+	}
+
+	glog.V(100).Infof("Creating server DPDK deployment %s in namespace %s",
+		serverDPDKDeploymentName, deploymentNamespace)
+
+	serverPodNetConfig := pod.StaticIPAnnotationWithMacAndNamespace(
```

Comment:
as a precaution I suggest adding variation that _serverPodNetConfig_ is neither empty nor _nil_